### PR TITLE
Get query groups

### PR DESCRIPTION
**Pull request by twrzes**

### CODE REVIEW

**Comment by twrzes** [comment]
Hi @kmalisz ,
I just pushed the recent changes to the branch. As you will see, I left the parameters and workflow definition in the script, if you want to test this module.

I have not tested @danielwojciechowskiardigen script extensively, therefore we may want to do that before merging this branch to the `dev` branch.

Let me know what you think!

File: bin/get_query_groups.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import pandas as pd
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-query', help='query file path', type=str)
+    parser.add_argument('-list-of-targets', help='target file path', type=str,  nargs='*')
+    parser.add_argument('-align', default='h_v', type=str, nargs='*')
+    parser.add_argument('-match', default='h_v h_cdr3_len', type=str, nargs='*')
```
Comment (kmalisz) [inline bin/get_query_groups.py]: Default values for this two might be confusing in my opinion.

File: bin/get_query_groups.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    parser.add_argument('-output', type=str)
+    parser.add_argument('--output-query-fasta', default='output_query_fasta.csv')
```
Comment (kmalisz) [inline bin/get_query_groups.py:16]: Output file should be in *.fasta format not *.csv

File: modules/local/get_query_groups.nf [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    val match
+
+    output:
+    path 'fasta_dir/*/*.fasta', emit: fasta_dir
+    path 'query_fasta.csv', emit: query_fasta_csv
+    path 'ref_paths.csv', emit: ref_paths_csv
+
+
+    script:
+    list_of_reference_files_space_separated = list_of_reference_files.join(' ')
```
Comment (kmalisz) [inline modules/local/get_query_groups.nf:27]: I'm not sure if this one is needed. I've never used this conversion usually .collect() on channel is enough to have correct nargs+ :D

**Review by kmalisz — CHANGES_REQUESTED** [review]

File: bin/get_query_groups.py (hunk 1 above)
Comment (twrzes) [reply bin/get_query_groups.py]: Do you want to remove these two options? @WojciechowskiDaniel what do you think?

File: bin/get_query_groups.py (hunk 2 above)
Comment (twrzes) [reply bin/get_query_groups.py:16]: This output is a file containing group_id and path to the `query.fasta`. Do you want to name it differently? @WojciechowskiDaniel , what do you think?

File: modules/local/get_query_groups.nf (hunk 3 above)
Comment (twrzes) [reply modules/local/get_query_groups.nf:27]: Not really, be aware that we need a space-separated string, not a list (which is an output of `.collect()` method).

File: bin/get_query_groups.py (hunk 2 above)
Comment (kmalisz) [reply bin/get_query_groups.py:16]: this one can be ignored after speaking with Tomek

File: bin/get_query_groups.py (hunk 1 above)
Comment (twrzes) [reply bin/get_query_groups.py]: Removed default values

**Pull request merged**